### PR TITLE
[FIXED JENKINS-28643] - Event times are being logged in seconds

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Seems I've read a wrong doc on docker APIs. All modern ones return times in seconds.

@reviewbybees @jtnord

### CODE REVIEW

**Comment by tfennelly** [comment]
:+1: 

But who decided to operate in seconds? :)

**Comment by oleg-nenashev** [comment]
@tfennelly 
Docker guys, I suppose. They track events in seconds, but publish all other times as **nano**second(!) strings. What a mess :(

File: src/main/java/org/jenkinsci/plugins/docker/traceability/core/DockerTraceabilityRootAction.java [hunk 1]
```diff
@@ -242,7 +242,7 @@ public HttpResponse doSubmitContainerStatus(
         checkPermission(DockerTraceabilityPlugin.SUBMIT);
         final ObjectMapper mapper = new ObjectMapper();
         final InspectContainerResponse[] inspectContainerResponses = mapper.readValue(inspectData, InspectContainerResponse[].class);
-        final long eventTime = time != 0 ? time : new Date().getTime();
+        final long eventTime = time != 0 ? time : new Date().getTime()/1000;
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/core/DockerTraceabilityRootAction.java]: `System.currentTimeMillis()` would avoid the temporayObject creation.
Could also Use `java.util.concurrent.TimeUnit`s conversion to make it more explicit.

File: src/main/java/org/jenkinsci/plugins/docker/traceability/fingerprint/DockerInspectImageFacet.java [hunk 2]
```diff
@@ -68,7 +68,7 @@ public DockerInspectImageFacet(@Nonnull Fingerprint fingerprint, long timestamp,
     
     /**
      * Time, when the latest report has been submitted.
-     * @return The time is specified in milliseconds since January 1, 1970, 00:00:00 GMT
+     * @return The time is specified in seconds since January 1, 1970, 00:00:00 GMT
      */
     public long getReportTimestamp() {
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/fingerprint/DockerInspectImageFacet.java]: I think internally we should be using normal java timestamps, or we should use Dates.
Someone somehere will get a conversion wrong.

**Comment by jtnord** [comment]
Can we not parse what we are given to a Date Object or to a normal java timestamp to prevent lots of conversion bugs "forget/does it wrong"?

File: src/main/java/org/jenkinsci/plugins/docker/traceability/fingerprint/DockerInspectImageFacet.java (hunk 2 above)
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/fingerprint/DockerInspectImageFacet.java]: No opinion. Both approaches are OK for me

File: src/main/java/org/jenkinsci/plugins/docker/traceability/core/DockerTraceabilityRootAction.java (hunk 1 above)
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/core/DockerTraceabilityRootAction.java]: Agreed

**Comment by oleg-nenashev** [comment]
@jtnord 

> Can we not parse what we are given to a Date Object or to a normal java timestamp to prevent lots of conversion bugs "forget/does it wrong"?

We still have seconds in the raw <code>Event</code> field of <code>DockerTraceabilityReport</code>. We cannot change this field, so I would not change the API.

What about renaming the  <code>DockerInspectImageFacet::getReportTimestamp()</code> method to <code>DockerInspectImageFacet::getReportTimeInSeconds()</code>? It is as horrible as obvious

**Comment by jtnord** [comment]
@oleg-nenashev - anything that makes the API more obvious that its not a java timestamp time is good.
So getReportTimeInSeconds() works for me.

**Comment by oleg-nenashev** [comment]
@jtnord 
Done. Also added some Javadocs to the Facet

**Comment by jtnord** [comment]
:+1:

**Pull request merged**